Tests/Tests/DDLogFileManagerTests.m

```diff
@@ -0,0 +1,154 @@
+// Software License Agreement (BSD License)
+//
+// Copyright (c) 2010-2018, Deusty, LLC
+// All rights reserved.
+//
+// Redistribution and use of this software in source and binary forms,
+// with or without modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+//
+// * Neither the name of Deusty nor the names of its contributors may be used
+//   to endorse or promote products derived from this software without specific
+//   prior written permission of Deusty, LLC.
+
+@import XCTest;
+#import <CocoaLumberjack/CocoaLumberjack.h>
+
+
+#pragma mark Test Case for DDLogFileManagerDefault
+// Feel free to extend :-)
+
+@interface DDLogFileManagerDefaultTests : XCTestCase
+@property (nonatomic, strong, readwrite) DDLogFileManagerDefault *logFileManager;
+@end
+
+@implementation DDLogFileManagerDefaultTests
+
+- (void)setUp {
+    [super setUp];
+    [self setUpLogFileManager];
+}
+
+- (void)setUpLogFileManager {
+    NSString *logsDirectory = NSTemporaryDirectory();
```

Comment:
Please, use `__auto_type` instead.

Tests are creatures from the modern world.

Reply:
Could you please explain to me what the benefit of using `__auto_type` in this case is?
It uses up more characters, does not explicitly tell a reader what the type is and, I assume, the compiled code will be the same. I can understand wanting to use `__auto_type` to infer the type when the resulting type is complex but here it is very simple.
I don't write Objective-C on an everyday basis so I'm just being curious here :-)

Reply:
@tmsrtx 

It is not the only one place, I just choose random place.

You have more "simple type" example.

```objective-c
        unsigned long long fileSize = [[[NSFileManager defaultManager] attributesOfItemAtPath:filePath error:nil][NSFileSize] unsignedLongLongValue];
```

```objective-c
        __auto_type fileSize = [[[NSFileManager defaultManager] attributesOfItemAtPath:filePath error:nil][NSFileSize] unsignedLongLongValue];
```

And, of course, writing mantras like: `NSString *something = [[NSString alloc] init..]` is not the best style.

What about top-level functions and this example?

```objective-c
    NSString *logsDirectory = NSTemporaryDirectory();
```

It is already incorrect. 

```objective-c
NSTemporaryDirectory(); // return type nonnull NSString *
```

For example, if `nullability` can become a part of type ( or any type placeholder construction ), my advise will take zero efforts.